backend/src/main/scala/com/softwaremill/bootzooka/user/UserModel.scala

```diff
@@ -50,17 +50,33 @@ class UserModel {
 }
 
 case class User(
-    id: Id @@ User,
-    login: String,
-    loginLowerCased: String @@ LowerCased,
-    emailLowerCased: String @@ LowerCased,
-    passwordHash: PasswordHash[SCrypt],
-    createdOn: Instant
+                 id: Id @@ User,
+                 login: String,
+                 loginLowerCased: String @@ LowerCased,
+                 emailLowerCased: String @@ LowerCased,
+                 passwordHash: String @@ Hashed,
+                 createdOn: Instant
 ) {
 
-  def verifyPassword(password: String): VerificationStatus = SCrypt.checkpw[cats.Id](password, passwordHash)
+  def verifyPassword(password: String): PasswordVerificationStatus =
+    if (Password.check(password, passwordHash) `with` ARGON_2) Verified else VerificationFailed
 }
 
 object User {
-  def hashPassword(password: String): PasswordHash[SCrypt] = SCrypt.hashpw[cats.Id](password)
+  object PasswordHashing {
+
+    val ARGON_2: Argon2Function = Argon2Function.getInstance(
```

Comment:
[minor] in Scala the convention for constants is `Argon2`, `MemoryInKib` etc. (starts with a captial letter)

Reply:
Updated, thanks!